### PR TITLE
New DAG for airnow

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
**_WHAT DOES THIS PR DO?_**
- Creating DAG to restream US embassy data

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []()
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

File: src/airflow/dags/airnow.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    @task()
+    def extract_bam_data():
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+        from airqo_etl_utils.date import DateUtils
+
+        start_date_time, end_date_time = DateUtils.get_query_date_time_values()
+
+        return AirnowDataUtils.extract_bam_data(
+            organisation= "usembassy", start_date_time=start_date_time, end_date_time=end_date_time
+        )
```
Comment (NoahNsimbe) [inline src/airflow/dags/airnow.py:69]: Hi @BenjaminSsempala, great work!. 
US embassy seems hardcoded. Could we make this DAG dynamic (not tied to US Embassy only)??

Comment (BenjaminSsempala) [reply src/airflow/dags/airnow.py]: @NoahNsimbe I've made these changes ahead of our sync today

File: src/airflow/airqo_etl_utils/airnow_api.py [hunk 2]
```diff
@@ -56,6 +61,16 @@ def __request(self, endpoint, params):
...
+        
+    def get_networks(self) ->List[str]:
+        # TODO: Create endpoint to return networks
+        return [
+            {
+                "network":Tenant.US_EMBASSY,
+                "data_source":DataSource.AIRNOW,
+                "api_key":self.USEMBASSY_API_KEY,
+            }
+        ]
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/airnow_api.py:73]: @BenjaminSsempala the return type does not match what you are returning.

File: src/airflow/dags/airnow.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                    api_key=network["api_key"],
+                    start_date_time=start_date_time,
+                    end_date_time=end_date_time,
+                )
+                extracted_bam_data = pd.concat(
+                    [extracted_bam_data, network_data], ignore_index=True
+                )
+
+        return extracted_bam_data
+
```
Comment (NoahNsimbe) [inline src/airflow/dags/airnow.py:43]: @BenjaminSsempala how are you differentiating data for the various networks??

File: src/airflow/dags/airnow.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    @task()
+    def process_data(airnow_data: pd.DataFrame):
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+
+        return AirnowDataUtils.process_bam_data(data=airnow_data)
```
Comment (NoahNsimbe) [inline src/airflow/dags/airnow.py:48]: @BenjaminSsempala process_bam_data requires two positional arguments but you provided one. How will this task run successfully??

File: src/airflow/dags/airnow.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+        return extracted_bam_data
+
+    @task()
+    def process_data(data: pd.DataFrame):
+        from airqo_etl_utils.airnow_utils import AirnowDataUtils
+
+        processed_data =pd.DataFrame()
+        networks = AirNowApi().get_networks()
+        for network in networks:
```
Comment (NoahNsimbe) [inline src/airflow/dags/airnow.py:106]: This looks like a repetition. The first tasks already attached the network/tenant @BenjaminSsempala

File: src/airflow/airqo_etl_utils/message_broker_utils.py [hunk 6]
```diff
@@ -87,18 +87,18 @@ def update_hourly_data_topic(data: pd.DataFrame):
         ]
         devices.rename(
             columns={
+                "device_id": "device_name",
                 "mongo_id": "device_id",
-                "name": "device_name",
                 "latitude": "device_latitude",
                 "longitude": "device_longitude",
             },
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/message_broker_utils.py:94]: @BenjaminSsempala this function needs to be simplified

**Pull request merged**